raft/src/test/java/org/apache/kafka/raft/QuorumStateTest.java

```diff
@@ -475,4 +486,31 @@ public void testObserverDetachLeader() throws IOException {
         assertEquals(1, state.epoch());
     }
 
-}
\ No newline at end of file
+    @Test
+    public void testInitializeWithCorruptedStore() throws IOException {
```

Comment:
We can simplify this test case and remove the dependence on `FileBasedStateStore` by adding the ability to raise exceptions from `MockQuorumStateStore`.